File: WWF/WF_WCF_Samples v4.0Beta1/WF/Basic/CustomActivities/CustomActivityDesigners/WorkflowItemPresenter/vb/SimpleNativeActivity.vb
'-------------------------------------------------------------------
' Copyright (c) Microsoft Corporation. All rights reserved
'-------------------------------------------------------------------

Imports System.Activities

Public Class SimpleNativeActivity
    Inherits NativeActivity

    Public Property Body As WorkflowElement

    Protected Overloads Overrides Sub Execute(ByVal context As System.Activities.ActivityExecutionContext)
        context.ScheduleActivity(Body)
    End Sub
End Class

File: WWF/WF_WCF_Samples v4.0Beta1/WF/Basic/CustomActivities/CustomActivityDesigners/WorkflowItemPresenter/cs/SimpleNativeActivity.cs
//-------------------------------------------------------------------
// Copyright (c) Microsoft Corporation. All rights reserved
//-------------------------------------------------------------------

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Activities;

namespace Microsoft.Samples.UsingWorkflowItemPresenter
{
    public sealed class SimpleNativeActivity : NativeActivity
    {
        public WorkflowElement Body { get; set; }

        protected override void OnGetChildren(IList<WorkflowElement> children)
        {
            children.Add(Body);
        }

        protected override void Execute(ActivityExecutionContext context)
        {
            context.ScheduleActivity(Body);
        }
    }
}
